src/main/java/com/artipie/rpm/Rpm.java

```diff
@@ -160,47 +143,7 @@ public Completable update(final String key) {
      * @return Completion or error signal.
      */
     public Completable update(final Key key) {
-        return Single.fromCallable(() -> Files.createTempFile("rpm", ".rpm"))
-            .flatMap(
-                temp -> new RxFile(temp, this.vertx.fileSystem())
-                    .save(
-                        new RxStorageWrapper(this.storage).value(key)
-                            .flatMapPublisher(pub -> pub)
-                    )
-                    .andThen(Single.just(new Pkg(temp))))
-            .flatMapCompletable(
-                pkg -> {
-                    final Repomd repomd = new Repomd(
-                        this.storage,
-                        this.vertx,
-                        this.naming,
-                        this.dgst
-                    );
-                    return Completable.concatArray(
-                        repomd.update(
-                            "primary",
-                            new SynchronousAct(
-                                file -> new Primary(file, this.dgst).update(key, pkg),
-                                this.primary
-                            )
-                        ),
-                        repomd.update(
-                            "filelists",
-                            new SynchronousAct(
-                                file -> new Filelists(file, this.dgst).update(pkg),
-                                this.filelists
-                            )
-                        ),
-                        repomd.update(
-                            "other",
-                            new SynchronousAct(
-                                file -> new Other(file, this.dgst).update(pkg),
-                                this.other
-                            )
-                        )
-                    );
-                }
-            );
+        return this.batchUpdate(key);
```

Comment:
@brastak it looks not correct: `update(key)` accepts `key` as a reference to RPM package, but `batchUpdate` accepts `key` as a repository directory with all RPM packages.

Reply:
Both of the works the same way now: looks for ALL rpms in the storage and processing them to create metadata. batchUpdate does not use passed key now as you can see. I have a plans to optimize it but it's quite tricky XML manipulations on-the-fly